### ISSUE TITLE
Bump etcd version to 3.5.8 for supported K8s releases

### ISSUE THREAD

**Issue by pchan** [kubernetes/kubernetes#117648]
### What happened?

etcd 3.5.6 is being used for K8s [1.26](https://github.com/kubernetes/kubernetes/blob/f89670c3aa4059d6999cb42e23ccb4f0b9a03979/build/dependencies.yaml#L56-L58) and [1.25](https://github.com/kubernetes/kubernetes/blob/a1a87a0a2bcd605820920c6b0e618a8ab7d117d4/build/dependencies.yaml#L56-L58). It contains a number CVE vulnerabilities that have been fixed in subsequent patch version (v3.5.7, v3.5.8), mostly via move to distroless .  Specifically trivy shows 33 High and 8 critical fixes in v3.5.6 while it zero in v3.5.8. The full list of critical CVE issues in 3.5.6, from trivy, is here [1]. For High CVEs issues the trivy command [2] can be run.

See the [3.5.8 changelog](https://github.com/etcd-io/etcd/blob/main/CHANGELOG/CHANGELOG-3.5.md#v358-tbd)  and [3.5.7 changelog](https://github.com/etcd-io/etcd/blob/main/CHANGELOG/CHANGELOG-3.5.md#v357-2023-01-20)  for more details on etcd releases.

There is already an attempt to move main to etcd 3.5.8 by PRs https://github.com/kubernetes/kubernetes/pull/117332 , https://github.com/kubernetes/kubernetes/pull/117335, https://github.com/kubernetes/k8s.io/pull/5157

cc: @serathius , @nikhita 

[1] 
CVE-2022-1664
CVE-2019-8457
CVE-2022-1586
CVE-2022-1587
CVE-2022-1292
CVE-2022-2068
CVE-2021-46848
CVE-2022-37434

[2]
`trivy image --severity HIGH gcr.io/etcd-development/etcd:v3.5.6 `

### What did you expect to happen?

Move K8s 1.26 and 1.25 to etcd latest patch version v3.5.8.  

### How can we reproduce it (as minimally and precisely as possible)?

N/A

### Kubernetes version

<details>

```console
$ kubectl version
N/A```

</details>


### Cloud provider

<details>
N/A
</details>


### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by serathius** [comment]
I don't think the etcd image used in k/k repo is the authoritative image for which etcd version you should be using. It's just used for K8s testing. Instead of taking on maintenance burden to keep it up to date, I would rather convince people that they should use latest image.

**Comment by pchan** [comment]
> Instead of taking on maintenance burden to keep it up to date

Can you please elaborate on this ? The bump itself is a bunch of code changes. It has been backported before too (https://github.com/kubernetes/kubernetes/pull/114093). 

>  I would rather convince people that they should use latest image.

Agree with this, but having offical k8s against the latest etcd sets a better example.

**Comment by BenTheElder** [comment]
> I don't think the etcd image used in k/k repo is the authoritative image for which etcd version you should be using. It's just used for K8s testing. Instead of taking on maintenance burden to keep it up to date, I would rather convince people that they should use latest image.

erm, the image referenced by kubeadm matters at least. cc @neolit123

**Comment by neolit123** [comment]
there is an open PR to update kubeadm to 3.5.8:
https://github.com/kubernetes/kubernetes/pull/117335

etcd patch level backports to k8s patch levels in supported minor versions generally get my +1, but it's up to release team and other approvers whether they will go in.

**Comment by pacoxu** [comment]
https://github.com/kubernetes/kubernetes/pull/118027 etcd v3.5.9 for v1.28. So v1.29 is with 3.5.9.

The cherry-pick PRs are still not merged. 
- v1.27 is using etcd v3.5.7.
- v1.26 is using etcd v3.5.6.
- v1.26 is using etcd v3.5.5.

**Comment by jmhbnz** [comment]
Discussed during sig etcd triage meeting assigned @lavacat to check and close.

/assign @lavacat

**Comment by lavacat** [comment]
Tittle of this issue is obsolete. 
3.5.9 is already merged to 1.25, 1.26, 1.27, 1.28

3.5.10 is already merged into master and automated PRs are created:

[1.28](https://github.com/kubernetes/kubernetes/pull/121805)
[1.27](https://github.com/kubernetes/kubernetes/pull/121804)
[1.26](https://github.com/kubernetes/kubernetes/pull/121803)

They are approved by etcd team @jmhbnz

I think this issue can be closed

**Comment by jmhbnz** [comment]
Thanks @lavacat, makes sense & agree.

/close
